### PR TITLE
I added API documents.

### PR DESCRIPTION
**Pull request by dairyo**
***This PR is not workable but this PR is not merged to develop. Please feel free to merge this.***

I wrote some API documents. I fixed some APIs too.

To simply API, I fixed following APIs

* I fixed argument of completionHandlers for two 'query', 'count' and 'aggregate` methods.
* I fixed argument of `count`.
  * I changed an argument `field` to `target`. `target` is instance of `CountingTarget` class. I withdraw using tuple at this argument.
* I put together 'function' and `field` arguments to `aggregation` in `aggregate` method. 

Related PR: #265

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/Aggregation.swift [hunk 1]
```diff
@@ -8,16 +8,79 @@
...
+        /** A enumeration element to denote integer type. */
+        case integer = "INTEGER"
+
+        /** A enumeration element to denote decimal type. */
+        case decimal = "DECIMAL"
     }
+
+    // MARK: - Properties
+
+    /** A field name to be aggregated. */
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Aggregation.swift]: Name of the field to be aggregated.

File: ThingIFSDK/ThingIFSDK/Aggregation.swift [hunk 2]
```diff
@@ -8,16 +8,79 @@
...
+    open let name: String
+
+    /** A field type to be aggregated. */
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Aggregation.swift]: Type of the field of be aggregated.

File: ThingIFSDK/ThingIFSDK/Aggregation.swift [hunk 3]
```diff
@@ -8,16 +8,79 @@
...
+    open let type: FieldType
+
+    /** A function applied to aggregated fields. */
+    open let function: Function
+
+    // MARK: - Initializing TriggeredCommandForm instance.
+    /** Initializer of `Aggregation` instance.
+
+     - Parameter name: a field name to be aggregated.
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Aggregation.swift]: Name of field to be aggregated.

File: ThingIFSDK/ThingIFSDK/Aggregation.swift [hunk 4]
```diff
@@ -8,16 +8,79 @@
...
+     - Parameter type: a field type to be aggregated.
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Aggregation.swift]: Type of the field to be aggregated.

File: ThingIFSDK/ThingIFSDK/CountingTarget.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+import Foundation
+
+/**
+ A class to specify counting target.
+
+ This class contains data in order to count with
+`ThingIFAPI.count(_:_:_:clause:firmwareVersion:completionHandler:)`.
+ */
+open class CountingTarget: NSObject, NSCoding {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/CountingTarget.swift]: I think naming with `Target` makes confused with `Target` thing. What about `CountedField`

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 6]
```diff
@@ -737,72 +737,141 @@ open class ThingIFAPI: NSObject, NSCoding {
     }
 
     // MARK: Qeury state history.
+
+    /** Query history states with trait alias.
+
+     - Parameter alias: Target trait alias to query.
+     - Parameter clause: Clause to narrow down history states. if nil,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: `Clause` > `QueryClause`

File: ThingIFSDK/ThingIFSDK/CountingTarget.swift (hunk 5 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/CountingTarget.swift]: @yongpingchen 

Umm, `CountingTarget` has `FieldType` enum. So If we change to `CountedField`, The enum becomes `CountedField.FieldType`. This is my concern.

What do you think?

**Comment by dairyo** [comment]
@yongpingchen 

I fixed according to your comment. I also added reply to your comment. Please check it.

**Comment by dairyo** [comment]
@yongpingchen 

I fixed according to your comment.

**Pull request merged**